test/Microsoft.AspNetCore.Mvc.FunctionalTests/MvcEncodedTestFixtureOfT.cs

```diff
@@ -16,9 +16,9 @@ protected override void ConfigureApplication(MvcWebApplicationBuilder<TStartup>
             base.ConfigureApplication(builder);
             builder.ConfigureBeforeStartup(services =>
             {
-                services.AddTransient<HtmlEncoder, HtmlTestEncoder>();
-                services.AddTransient<JavaScriptEncoder, JavaScriptTestEncoder>();
-                services.AddTransient<UrlEncoder, UrlTestEncoder>();
+                services.TryAddTransient<HtmlEncoder, HtmlTestEncoder>();
```

Comment:
That's why they are added before startup configure services run. It tests that even if you do it this way it will work.